### PR TITLE
RHBRMS-2730, RHBRMS-2729

### PR DESCRIPTION
**Pull request by lazarotti**

### CODE REVIEW

**Comment by lazarotti** [comment]
@manstis, the only difference than that you showed me is adding the full stack trace also to logger.debug. Even adding the exception message to:

            final String msg = MessageFormat.format( ERROR_EXTERNAL_CLASS_VERIFICATION,
                                             toFQCN( packageName, className ),
                                             e.getMessage() );

... sometimes it is still not clear. In that jira even fixing the message it will show only "Underlying system error is Lme/lazarotti/SaySomething;" - however if added the stacktrace we will see in the stack: 
"Caused by: java.lang.ClassNotFoundException: me.lazarotti.SaySomething"

That's why I also added it.

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/builder/ClassVerifier.java [hunk 1]
```diff
@@ -76,9 +76,11 @@ private void verifyClass( final String packageName,
             }
 
         } catch (Throwable e) {
-            final String msg = MessageFormat.format( ERROR_EXTERNAL_CLASS_VERIFICATION,
-                                                     toFQCN( packageName, className ) );
+           final String msg = MessageFormat.format( ERROR_EXTERNAL_CLASS_VERIFICATION,
+                                             toFQCN( packageName, className ),
+                                             e.getMessage() );
```
Comment (hasys) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/builder/ClassVerifier.java]: Hi @lazarotti, this new formatting for the fast look like you are calling MessageFormat.*format*().toFQCN.().e.getMessage(), I'm not sure that this format is better, than previous.

Comment (lazarotti) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/builder/ClassVerifier.java]: It is important to catch and show the exception, or the user will not be able to identify what dependency/transitve dependency is missing. See the Jira https://issues.jboss.org/browse/RHBRMS-2730

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/builder/ClassVerifier.java [hunk 2]
```diff
@@ -76,9 +76,11 @@ private void verifyClass( final String packageName,
...
             logger.warn( msg );
+	    logger.debug( "This state is usually encountered when the Project references a class not on the classpath; e.g. in a Maven 'provided' scope or 'optional' dependency.", e);
```
Comment (lazarotti) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/builder/ClassVerifier.java]: ACK, i will send a new PR with tabulation fixed

**Comment by lazarotti** [comment]
@manstis I am trying to figure out someway to simulate an exception during the class checking. I can't mock this with mockito as it does not work over final classes (java.lang.Class).

Can I add PowerMockito to the test suite ? It allows that. To understand  what I am trying to do see the test below (which does not work with Mockito):
http://pastebin.test.redhat.com/445192 (sure, the assert I will change to match an expected message with the error).
.
There is any other way to load a problematic class for verification in our test suite instead of mocking ?

**Comment by manstis** [comment]
@lazarotti Hi, I have no problem with PowerMockito being introduced (we use it elsewhere already). However you could more simply mock ```kieModuleMetaData.getClass.getClass(..)``` to throw ```ClassNotFoundException``` - or at least that's what I was thinking!

**Comment by lazarotti** [comment]
Closing this one, please review the new PR https://github.com/droolsjbpm/kie-wb-common/pull/636

**Pull request closed without merging**